### ISSUE TITLE
Craft running validation despite $runValidation being set to false

### ISSUE THREAD

**Issue by codyjames** [craftcms/cms#7756]
### Description
I have some pretty complex custom validation that I am running on EVENT_BEFORE_VALIDATE. However, sometimes I need to skip validation and just let the element get saved regardless of whether it validates or not.

We've been running into issues where elements (products in this case) are getting their titles changed back to a default title ("Product 234882") when we are saving them with $runValidation set to false.

The led me to this block of code in `services/Elements.php`:
```   
// Set a dummy title if there isn't one already and the element type has titles
if (!$runValidation && $element::hasContent() && $element::hasTitles() && !$element->validate(['title'])) {
    if ($isNewElement) {
        $element->title = Craft::t('app', 'New {type}', ['type' => $element::displayName()]);
    } else {
        $element->title = $element::displayName() . ' ' . $element->id;
    }
}
```

This piece of code is odd because it runs validation (even when $runValidation is set to false) which then triggers my additional validation on EVENT_BEFORE_VALIDATE. This results in _any element with validation errors getting their title overwritten_.

### Steps to reproduce

1. Write some custom validation events that fire on EVENT_BEFORE_VALIDATE.
2. Run saveElement with runValidation set to false on an element that'd fail your custom validation.
3. See the title get overwritten to a default title.

### Additional info

- Craft version: 3.6.11.2
- PHP version: 7.4.15

**Comment by brandonkelly** [comment]
It’s essential that Craft ensures that elements have valid titles before saving, which is why we make an exception for validating the title even if `$runValidation` is `false`. But you’ve got a good point – that can have unexpected consequences if there’s custom validation going on via `EVENT_BEFORE_VALIDATE` or `EVENT_AFTER_VALIDATE`. Just fixed that for the next release.

**Comment by brandonkelly** [comment]
We just released Craft 3.6.12 with this change.
